drivers/serial/uart_rpi_pico.c

```diff
@@ -14,62 +15,307 @@
 
 struct uart_rpi_config {
 	uart_inst_t *const uart_dev;
-	uint32_t baudrate;
 	const struct pinctrl_dev_config *pcfg;
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_config_func_t irq_config_func;
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+struct uart_rpi_data {
+	uint32_t baudrate;
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_user_data_t irq_cb;
+	void *irq_cb_data;
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
 };
 
 static int uart_rpi_poll_in(const struct device *dev, unsigned char *c)
 {
-	const struct uart_rpi_config *config = dev->config;
+	const struct uart_rpi_config * const config = dev->config;
+	uart_inst_t * const uart_inst = config->uart_dev;
+	uart_hw_t * const uart_hw = uart_get_hw(uart_inst);
 
-	if (!uart_is_readable(config->uart_dev)) {
+	if (!uart_is_readable(uart_inst)) {
```

Comment:
why is this change needed? it doesn't seem related to interrupt support.

Reply:
This change is not needed, sorry for that.